pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -146,10 +146,12 @@ def restore_model_weights(self, checkpoint_path: Optional[Union[str, Path]]) ->
         """ Restore only the model weights. """
         checkpoint = self._loaded_checkpoint
         if checkpoint_path is not None:
-            checkpoint = self.trainer.training_type_plugin.load_checkpoint_file(checkpoint_path)
-
-        self.trainer.lightning_module.on_load_checkpoint(checkpoint)
-        self.trainer.training_type_plugin.load_model_state_dict(checkpoint)
+            if hasattr(self.trainer.training_type_plugin, "serialized_restore_model_state"):
```

Comment:
Just personal preference, but I would prefer to refactor the code and pass the `checkpoint_path` to the training_type plugin and it handles the loading logic.